packages/@n8n/client-oauth2/src/ClientOAuth2.ts

```diff
@@ -85,7 +85,9 @@ export class ClientOAuth2 {
 		if (query) {
 			url += (url.indexOf('?') === -1 ? '?' : '&') + query;
 		}
-
+		if (options?.headers?.Authorization === '') {
```

Comment:
did you check if this is actually needed? options might be generated by `utils.getRequestOptions`, [which already deletes this header](https://github.com/n8n-io/n8n/blob/master/packages/%40n8n/client-oauth2/src/utils.ts#L79).

Reply:
Ah yes indeed the options is being generated using the `utils.getRequestOptions` in the `codeFlow.ts`